plugin/manager.go

```diff
@@ -133,42 +145,37 @@ func (pm *Manager) tmpDir() string {
 	return filepath.Join(pm.config.Root, "tmp")
 }
 
-// StateChanged updates plugin internals using libcontainerd events.
-func (pm *Manager) StateChanged(id string, e libcontainerd.StateInfo) error {
-	logrus.Debugf("plugin state changed %s %#v", id, e)
-
-	switch e.State {
-	case libcontainerd.StateExit:
-		p, err := pm.config.Store.GetV2Plugin(id)
-		if err != nil {
-			return err
-		}
+// HandleExitEvent is called when the executor receives the exit event
+// In the future we may change this, but for now all we care about is the exit event.
+func (pm *Manager) HandleExitEvent(id string) error {
+	p, err := pm.config.Store.GetV2Plugin(id)
+	if err != nil {
+		return err
+	}
 
-		os.RemoveAll(filepath.Join(pm.config.ExecRoot, id))
+	os.RemoveAll(filepath.Join(pm.config.ExecRoot, id))
```

Comment:
`os.RemoveAll()` returns an error, maybe `logrus.Warn()` on error? This looks like it was an existing issue, so not blocking